pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -593,6 +593,21 @@ def _init_strategy(self) -> None:
         else:
             raise RuntimeError(f"{self.strategy} is not valid type: {self.strategy}")
 
+        from pytorch_lightning.utilities import _IS_INTERACTIVE
+
+        is_interactive_compatible = (
+            self.strategy.is_interactive_compatible if hasattr(self.strategy, "is_interactive_compatible") else False
```

Comment:
should these specific hasattr checks could be formalized as traits or mixins? someone has to go deep into the code to realize these specific properties need to be added to their strategy for this to work. we had the same discussion for backward sync